### ISSUE TITLE
why can't use batchnorm layer?

### ISSUE THREAD

**Issue by duzeyan** [caogang/wgan-gp#17]
if netD has batchnorm layer,pytorch will report error when run this code gradient_penalty.backward()

**Comment by caogang** [comment]
Do you use the latest pytorch version?

**Comment by LIU-Xuanqing** [comment]
@duzeyan Seems that pytorch 0.2.0 has a bug with autograd.gradient function, if the network contains batchnorm layer then gpu memory won't be released promptly.
